core/src/main/java/io/confluent/kafka/schemaregistry/rest/resources/SubjectVersionsResource.java

```diff
@@ -664,4 +673,32 @@ public void modifyTags(
     }
     asyncResponse.resume(registerSchemaResponse);
   }
+
+  private void modifyPreviousRuleSet(String subject, TagSchemaRequest request)
+      throws SchemaRegistryException {
+    int oldVersion = request.getNewVersion() != null ? request.getNewVersion() - 1 : -1;
+    Schema oldSchema = schemaRegistry.get(subject, oldVersion, false);
+    // Use the previous ruleSet instead of the passed in one
+    RuleSet ruleSet = oldSchema != null ? oldSchema.getRuleSet() : null;
+    if (request.getRulesToMerge() != null) {
+      ruleSet = mergeRuleSets(ruleSet, request.getRulesToMerge());
+    }
+    if (ruleSet != null && request.getRulesToRemove() != null) {
+      List<String> rulesToRemove = request.getRulesToRemove();
+      List<Rule> migrationRules = ruleSet.getMigrationRules();
+      if (migrationRules != null) {
+        migrationRules = migrationRules.stream()
+            .filter(r -> !rulesToRemove.contains(r.getName()))
+            .collect(Collectors.toList());
+      }
+      List<Rule> domainRules = ruleSet.getDomainRules();
+      if (domainRules != null) {
+        domainRules = domainRules.stream()
+            .filter(r -> !rulesToRemove.contains(r.getName()))
+            .collect(Collectors.toList());
+      }
+      ruleSet = new RuleSet(migrationRules, domainRules);
+    }
+    request.setRuleSet(ruleSet);
```

Comment:
What if user sets `ruleSet` and `ruleToMerge` field? It seems like the values in `ruleToMerge` would overwrite the values in `ruleSet`

Reply:
Sure, I'll add a check